fastparquet/converted_types.py

```diff
@@ -174,10 +166,7 @@ def from_bytes(d):
     return data
 
 
-@numba.njit(nogil=True)
-def time_shift(indata, outdata, factor=1000):  # pragma: no cover
-    for i in range(len(indata)):
-        if indata[i] == nat:
-            outdata[i] = nat
-        else:
-            outdata[i] = indata[i] * factor
```

Comment:
This is the problem line: Numpy on Win32 will use the `indata` `dtype='int32'` to perform the multiplication, causing overflow.